gencred/cmd/gencred/main.go

```diff
@@ -174,21 +272,132 @@ func Main() {
 		util.PrintErrAndExit(err)
 	}
 
-	loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
-	configOverrides := &clientcmd.ConfigOverrides{CurrentContext: o.context}
-	kubeconfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(loadingRules, configOverrides)
-
-	config, err := kubeconfig.ClientConfig()
-	if err != nil {
-		util.PrintErrAndExit(err)
+	var c config
+	if len(o.config) > 0 {
+		// Load from config yaml file
+		body, err := ioutil.ReadFile(o.config)
+		if err != nil {
+			util.PrintErrAndExit(err)
+		}
+		if err := yaml.Unmarshal(body, &c); err != nil {
+			util.PrintErrAndExit(err)
+		}
+	} else {
+		c.Clusters = []*clusterConfig{
+			{
+				Context:            &o.context,
+				Name:               o.name,
+				WithCertificate:    o.certificate,
+				WithServiceAccount: o.serviceaccount,
+				DurationInSeconds:  o.duration,
```

Comment:
Yeah I think it's weird too. The entire problem is that the yaml package used in this repo doesn't support duration. So I think it would be better change o.durationInSeconds to also be int64 to ensure consistency

Reply:
awesome, glad I saw this before working on the change :)

Reply:
done